authmate/authmate.go

```diff
@@ -119,13 +126,30 @@ func (a *Agent) checkContainer(ctx context.Context, cid *cid.ID, friendlyName st
 	return cid, nil
 }
 
-func (a *Agent) getCurrentEpoch(ctx context.Context) (uint64, error) {
+func (a *Agent) getEpochDurations(ctx context.Context) (*epochDurations, error) {
 	if conn, _, err := a.pool.Connection(); err != nil {
-		return 0, err
+		return nil, err
 	} else if networkInfo, err := conn.NetworkInfo(ctx); err != nil {
-		return 0, err
+		return nil, err
 	} else {
-		return networkInfo.CurrentEpoch(), nil
+		res := &epochDurations{
+			currentEpoch: networkInfo.CurrentEpoch(),
+			msPerBlock:   networkInfo.MsPerBlock(),
+		}
+
+		networkInfo.NetworkConfig().IterateParameters(func(parameter *netmap.NetworkParameter) bool {
+			if string(parameter.Key()) == "EpochDuration" {
+				data := make([]byte, 8)
+				copy(data, parameter.Value())
```

Comment:
Do we need to copy slice for `binary.LittleEndian.Uint64`?

Reply:
Yes. Otherwise it panics because `parameter.Value()` can have length less than 8